api/types.go

```diff
@@ -704,10 +707,11 @@ func (pT PinType) String() string {
 
 // PinOptions wraps user-defined options for Pins
 type PinOptions struct {
-	ReplicationFactorMin int    `json:"replication_factor_min"`
-	ReplicationFactorMax int    `json:"replication_factor_max"`
-	Name                 string `json:"name"`
-	ShardSize            uint64 `json:"shard_size"`
+	ReplicationFactorMin int               `json:"replication_factor_min" codec:"rn,omitempty"`
```

Comment:
@hsanjuan Is there a compile-time or test-time check that we can add so that we don't conflict with the codec variable?

Reply:
As discussed, tests break in most cases when screwing up the codec field names.

Reply:
Additionally I have added a test that verifies that there are no dup fields.